cli/src/main/java/com/devonfw/tools/ide/tool/intellij/Intellij.java

```diff
@@ -31,10 +67,31 @@ public boolean install(boolean silent) {
   }
 
   @Override
-  public void installPlugin(PluginDescriptor plugin) {
+  protected void postInstall() {
+
+    super.postInstall();
+    if (this.context.getSystemInfo().isMac()) {
+      setMacOsFilePermissions(getToolPath().resolve("IntelliJ IDEA CE.app").resolve("Contents").resolve("MacOS").resolve(IDEA));
+    }
+  }
 
-    // TODO Auto-generated method stub
+  private static void setMacOsFilePermissions(Path binaryFile) {
 
+    if (Files.exists(binaryFile)) {
+      String permissionStr = FileAccessImpl.generatePermissionString(493);
```

Comment:
Seems I missed to discuss on this API.
IMHO this is very bad design:
* numeric file permissions must always be given in octal notation or nobody can understand what `493` actually is. If you write it octal as `0755` at least some bash/unix experienced user will understand what this.
* We should avoid direct usage of `FileAccessImpl` in commandlets. That is why we have `FileAccess` interface.
* Why do we provide a method in FileAccessImpl that converts a number to a String and then we again convert the String to a `Set` of `PosixFilePermissions` in order to then natively set this via `Files.setPosixFilePermissions`? The idea of `FileAccess` is to create a higher level API and abstraction to such lower-level Java NIO operations and make our live easier. So if we believe that using numeric (octal) notation is the most readable way why dont we create a method in `FileAccess` to set file permissions to a given file with a given numeric value? However, it may be even simpler and more readable to simply define constants for the combinations that we need like this:
```
/** {@link PosixFilePermission}s for "rwxr-xr-x" or 0755. */
public static Set<PosixFilePermission> RWX_RX_RX = Set.of(PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE, PosixFilePermission.OWNER_EXECUTE, PosixFilePermission.GROUP_READ, PosixFilePermission.GROUP_EXECUTE, PosixFilePermission.OTHERS_READ, PosixFilePermission.OTHERS_EXECUTE);
```
* My real suggestion would rather be to add a method `FileAccess.makeExecutable(Path)` that will simply implement what `chmod a+x «path»` will do and could also be implemented to work on Windows (see also [here](https://stackoverflow.com/questions/664432/how-do-i-programmatically-change-file-permissions)).

Reply:
I've implemented your suggestions now.